src/data/sign.lean

```diff
@@ -0,0 +1,196 @@
+/-
+Copyright (c) 2022 Eric Rodriguez. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Eric Rodriguez
+-/
+import order.basic
+import algebra.algebra.basic
+import tactic.derive_fintype
+
+/-!
+# Sign function
+
+This file defines the sign function for types with zero and a decidable less-than relation, and
+proves some basic theorems about it.
+-/
+
+/-- The type of signs. -/
+@[derive [decidable_eq, inhabited, fintype]]
+inductive sign_type
+| zero | neg | pos
+
+namespace sign_type
+
+instance : has_zero sign_type := ⟨zero⟩
+instance :  has_one sign_type := ⟨pos⟩
```

Comment:
Or was aligning the functions like that on purpose?
```suggestion
instance : has_one  sign_type := ⟨pos⟩
```

Reply:
it was meant to be on purpose, but this looks better!